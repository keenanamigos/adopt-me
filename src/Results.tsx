import { FunctionComponent } from "react";
import Pet from "./Pet";
import { Pet as PetType } from './ApiResponseTypes';

interface Props {
    pets: PetType[];
}

const Results: FunctionComponent<Props> = ({ pets }) => {
    return (
        <div className="search">
            { !pets.length ? (
                <h2>No pets were found.</h2>
            ) : (
                pets.map(pet => (
                    <Pet
                        name={pet.name}
                        animal={pet.animal}
                        breed={pet.breed}
                        key={pet.id}
                        images={pet.images}
                        location={`${pet.city}, ${pet.state}`}
                        id={pet.id.toString()}
                    >
                    </Pet>
                ))
            )}
        </div>
    );
}

export default Results;