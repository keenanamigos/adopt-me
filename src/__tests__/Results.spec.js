import { describe, expect, test } from "@jest/globals";
import { create } from "react-test-renderer";
import Results from "../Results";

describe ("Results Component UI", () => {
    test("renders snapshot with no pets", () => {
        const tree = create(<Results pets={[]} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});