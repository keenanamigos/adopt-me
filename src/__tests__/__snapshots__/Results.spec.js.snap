// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Component UI renders snapshot with no pets 1`] = `
<div
  className="search"
>
  <h2>
    No pets were found.
  </h2>
</div>
`;
